docs/configuration.md

```diff
@@ -0,0 +1,124 @@
+is written to the database. If the update is more recent than `activity_resolution` seconds
+ago it's ignored. Increasing this value will reduce commits to the database.
+
+```yaml
+extraConfig:
+  myConfig: |
+    c.JupyterHub.activity_resolution = 6000
+```
+
+### `c.JupyterHub.last_activity_interval`
+This setting controls how often a periodic task in the hub named `update_last_activity`
+runs. This task updates user activity using information from the proxy. This task makes
+a large number of database calls and can put a fairly significant load on the hub. Zero to
+Jupyterhub sets this to 1 minute by default. The upstream default of 5 minutes is a better
+setting.
+
+```yaml
+extraConfig:
+  myConfig: |
+    c.JupyterHub.last_activity_interval = 300
+```
+
+### `JUPYTERHUB_ACTIVITY_INTERVAL`
+This controls how often each server reports its activity back to the hub. The default
+is 5 minutes and with hundreds or thousands of users posting activity updates it puts
+a heavy load on the hub and the hub's database. Increasing this to one hour or more
+reduces the load placed on the hub by these activity updates.
+
+```yaml
+singleuser:
+  extraEnv:
+    JUPYTERHUB_ACTIVITY_INTERVAL: "3600"
+```
+
+## Startup time
+
+### `init_spanwners_timeout`
+`c.JupyterHub.init_spawners_timeout` controls how long the hub will wait for spawners to
+initialize. When this timeout is reached the spawner check will go into the background and
+hub startup will continue. With many hundreds or thousands of spawners this is always going
+to exceed any reasonable timeout so there's no reason to wait at all. Setting it to `1` 
+(which is the minimum value) allows the hub to start faster and start servicing other requests.
+
+In `values.yaml`:
+```yaml
+extraConfig:
+  myConfig: |
+     c.JupyterHub.init_spawners_timeout = 1
+```
+
+## Other settings
+Other settings which are helpful for tuning performance.
+
+### `c.KubeSpawner.k8s_api_threadpool_workers`
+This value controls the number of threads `kubespawner` will create to make API calls to
+Kubernetes. The default is `5 * num_cpus`. Given a large enough number of users logging in
+and spawning servers at the same time this may not be enough threads. A more sensible value
+for this setting is `c.Jupyterhub.concurrent_spawn_limit`. `concurrent_spawn_limit` controls
+how many users can spawn servers at the same time. By creating that many threadpool workers
+we ensure that there's always a thread available to service a user's spawn request.
+
+In `values.yaml`:
+```yaml
+extraConfig:
+  perfConfig: |
+     c.KubeSpawner.k8s_api_threadpool_workers = c.JupyterHub.concurrent_spawn_limit
+```
+
+### Disable user events
+With this enabled `kubespawner` will process events from the Kubernetes API which are then
+used to show progress on the user spawn page. Disabling this reduces the load on `kubespawner`.
+
+To disable user events update the `events` key in the `values.yaml` file.
+```yaml
+singleuser:
+  events: false
+```
```

Comment:
Oh what do you think about having a generic `## References` section for linking to forum threads or other docs / videos, etc?

https://github.com/IBM/jupyter-tools/issues/3#issuecomment-686572133